### ISSUE TITLE
I can't run org-roam--build-cache-async

### ISSUE THREAD

**Issue by hackmalife** [jethrokuan/org-roam#131]
### Description

When I run org-roam--build-cache-async, error raises, the log is as follows:

`error in process sentinel: async-handle-result: Symbol’s function definition is void: org-roam--build-cache
error in process sentinel: Symbol’s function definition is void: org-roam--build-cache
e`

When I debug, it says

`Debugger entered--Lisp error: (void-function org-roam--build-cache)
  signal(void-function (org-roam--build-cache))
  async-handle-result((closure (t) (cache) (setq org-roam-forward-links-cache (plist-get cache :forward)) (setq org-roam-backward-links-cache (plist-get cache :backward)) (setq org-roam-titles-cache (plist-get cache :titles)) (setq org-roam-cache-initialized t) (message "Org-roam cache built!")) (async-signal (void-function org-roam--build-cache)) #<buffer *emacs*>)
  async-when-done(#<process emacs> "finished\n")`

When I run deft, it displays nothing, the warning in minibuffer is like this:
`user-error: The Org-Roam caches aren’t built! Please run org-roam--build-cache-async
`

#### Steps to Reproduce

1. Load emacs (I run emacs daemon and then emacsclient)
2-1. Run `org-roam--build-cache-async` (I did not put it in hook)
2-2. Run `deft`
...
-->

### Versions
- Emacs (`C-h v emacs-version`): v28.0.50
- Org-roam commit: https://github.com/jethrokuan/org-roam/commit/develop

**Comment by jethrokuan** [comment]
how did you install `org-roam`? Are your dotfiles somewhere?

**Comment by hackmalife** [comment]
it is pretty much same as README

(use-package org-roam
  :straight (:host github :repo "jethrokuan/org-roam" :branch "develop")
  :after org
  :hook
  (org-mode . org-roam-mode)
  :config
  (gsetq org-roam-directory (cloud-dir "org")))

**Comment by jethrokuan** [comment]
what's `gsetq`? also `org-roam-directory` should be a string, not a list.

I think if you placed a `(require 'org-roam-utils)` in `:config` you should be fine.

**Comment by hackmalife** [comment]
Oh, I use another function to return string there I just misrepresented it while I substitute it to report here. I confirmed that `org-roam-directory` has correct string path.
`gsetq` is just wrapper of `setq` in `general.el` package, it functions same.
Anyways, I just changed it to `setq` and confirmed that it yields the same log.
I placed `(require 'org-roam-utils)`, but result was same.
I tried to manually run `(org-roam--build-sync org-roam-directory)` and it kinda works, but even after that, when I try to run `org-roam--build-async` it yields the same error logs

`error in process sentinel: async-handle-result: Symbol’s function definition is void: org-roam--build-cache`
`error in process sentinel: Symbol’s function definition is void: org-roam--build-cache`
